enterprise/frontend/src/metabase-enterprise/database_replication/DatabaseReplicationForm.unit.spec.tsx

```diff
@@ -2,9 +2,20 @@ import userEvent from "@testing-library/user-event";
 
 import { renderWithProviders, screen, waitFor } from "__support__/ui";
 import type { PreviewDatabaseReplicationResponse } from "metabase-enterprise/api/database-replication";
+import {
+  useCreateDatabaseReplicationMutation,
+  usePreviewDatabaseReplicationMutation,
+} from "metabase-enterprise/api/database-replication";
 import { createMockDatabase } from "metabase-types/api/mocks";
 
 import { DatabaseReplicationForm } from "./DatabaseReplicationForm";
+import { DatabaseReplicationModal } from "./DatabaseReplicationModal";
+
+// Mock the API hooks for modal tests
+jest.mock("metabase-enterprise/api/database-replication", () => ({
+  useCreateDatabaseReplicationMutation: jest.fn(),
+  usePreviewDatabaseReplicationMutation: jest.fn(),
+}));
```

Comment:
If at all possible, we should not mock modules, but set up a fetch-mock mock response, which is the pattern used in the overwhelming majority of jest tests.

see reasoning here: https://github.com/metabase/metabase/pull/57481#discussion_r2074461844